yantra/manager.py

```diff
@@ -24,8 +24,19 @@ class PluginContainer(object):
 
     def __init__(self, plugin_type):
         self._plugins = []
+        self._errors = {}
         self.plugin_type = plugin_type
 
+    @property
+    def errors(self):
```

Comment:
It'll need to try to load the plugins to know if there are errors. Otherwise it'll have to rely on the client to call get_plugins first to know if there are errors. I agree that it seems weird-- we could separate the loading into two parts (pre-load and actual load/register) and call the pre-load part in the check.